state/application.go

```diff
@@ -272,10 +272,6 @@ func (a *Application) removeOps(asserts bson.D) ([]txn.Op, error) {
 			Id:     a.doc.DocID,
 			Assert: asserts,
 			Remove: true,
-		}, {
```

Comment:
Removed this and the other related bits. Will propose separately :) I do not think that it is particularly hurtful but was very surprising to examine a transaction where several exact same operations appeared...